lib/metadata.sh

```diff
@@ -19,6 +19,20 @@ meta_create() {
   kv_clear "$BUILD_DATA_FILE"
 }
 
+# initialize the metadata store, but do not modify it
+meta_init() {
```

Comment:
There are probably better names for these functions

Reply:
`read_only` is better, but it isn't entirely accurate since you could still write to the store 🤔 

I'll try splitting the logic that does the previous / current build swap into it's own separate function, and call that explicitly. That way `init` is always non-modifying